### ISSUE TITLE
Bootstrap using downgradeModule can't find AngularJS injector 

### ISSUE THREAD

**Issue by DavidMMelin** [ui-router/angular-hybrid#150]
I followed [this demo](https://stackblitz.com/edit/github-ypge8f-jmshp3) almost exactly as mentioned in [this issue](https://github.com/ui-router/angular-hybrid/issues/98), and can't get the Angular app to downgrade correctly without throwing the below error. 

```javascript
zone.js:672 Unhandled Promise rejection: Trying to get the AngularJS injector before it being set. ; Zone: <root> ; Task: Promise.then ; Value: Error: Trying to get the AngularJS injector before it being set.
```
The error fires when the Angular app is bootstrapped and the **UIRouterUpgradeModule** is loaded. It's almost like it can't find a reference to the AngularJS app (which is bootstrapped first), but I have no idea where to go from here. Has anyone had a similar issue?

**Comment by DavidMMelin** [comment]
For some reason a .catch block on the downgrading of the angular AppModule caused this issue. It was working fine before, but just started complaining when we tried to route though the Angular side.

**Comment by christopherthielen** [comment]
@DavidMMelin please see the discussion in https://github.com/ui-router/angular-hybrid/issues/93

I know of two common ways for the `Trying to get the AngularJS injector before it being set` message:

1) Multiple `UIRouterUpgradeModule.forChild()` calls.  (The sample app uses `UIRouterModule.forChild() in the feature and lazy modules)

This is addressed in version 6.0.0 by requiring `.forRoot()` to be used in the root and `.forChild()` to be used in feature modules and lazy modules.

2) Trying to bootstrap an Angular component as the root component (instead of an AngularJS component).  See my [explanation and stackblitz examples](https://github.com/ui-router/angular-hybrid/issues/93#issuecomment-390434817)

Hope your upgrade is going well.
